Resolves Issue #240 - Gsl1Fitter has issues with segmentation faults

spaulaus opened the pull request:
This resolves issue #240. 

The vectors that hold the data and weights was not being deleted before returning. This caused the segmentation fault on the second time through the Analyze method. 

In addition, the vector that held the fitted parameters was being freed before we actually got the parameter from it. This should resolve all the known issues with the Gsl1Fitter class.

spaulaus committed:
Resloves Issue #240

The vectors that hold the data and weights was not being deleted before returning. This caused the segmentation fault on the second time through the Analyze method. 

In addition, the vector that held the fitted parameters was being freed before we actually got the parameter from it. This should resolve all the known issues with the Gsl1Fitter class.

tking53 reviewed (changes requested):
Code compiles on kqxhc (see below). However it fails its unittest. 
```
/home/tking/programs/paass/Analysis/Resources/tests/unittest-GslFitter.cpp:25:1: error: Failure in TestGslFitter: Expected 0.85658 +/- 0.1 but was 6.90692e-310
FAILURE: 1 out of 1 tests failed (1 failures).
Test time: 0.00 seconds.

```
```
Linux Flavor
CentOS Linux release 7.3.1611 (Core)  

Linux Kernel 
3.10.0-514.6.1.el7.x86_64 

cmake version 2.8.12.2 

gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-11)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 

GSL version 
1.15
```

spaulaus committed:
Removing amplitude check in the GslFitter unittest

We do not really need this check since what we care about is that the phase is behaving properly.

spaulaus commented:
@tking53 : I have removed the unit test's check on the amplitude. We do not really care if the amplitude is different than what we thought it was going to be. In addition, the Gsl1Fitter no longer sets the amplitude variable, since it was never used for anything in the analysis.

tking53 commented:
Code compiles and runs on kqxhc (See my comment above for environment) the Gsl1Fitter now runs. The code produces the vandle histograms (3242-3245). Compared against the Gsl2Fitter these histograms are very similar (small fluctuation in the gamma flash centroid bin for 3245 which is fixable with a recalibration).
